src/main/ipc.js

```diff
@@ -116,6 +117,10 @@ function init (cwd, state) {
     dc.updateSettings(saved)
   })
 
+  ipc.on('retrieveAdvancedSettingsObj', (e, ...args) => {
+    windows.main.send('retrieveAdvancedSettingsObjResp', dc.getAdvancedSettings())
```

Comment:
Does it have to be this complicated and handled in a special way?

Reply:
we can do instead ipcRenderer(`dispatchSync`, `getAdvancedSettings`)